cognite/replicator/files.py

```diff
@@ -0,0 +1,215 @@
+import logging
+import mimetypes
+import time
+from typing import Dict, List
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes import FileMetadata
+from cognite.client.exceptions import CogniteAPIError
+
+from . import replication
+
+
+def create_file(
+    src_file: FileMetadata, src_dst_ids_assets: Dict[int, int], project_src: str, runtime: int
+) -> FileMetadata:
+    """
+    Makes a new copy of the file to be replicated based on a source file.
+
+    Args:
+        src_file: The file from the source to be replicated to destination.
+        src_dst_ids_assets: A dictionary of all the mappings of source asset id to destination asset id.
+        project_src: The name of the project the object is being replicated from.
+        runtime: The timestamp to be used in the new replicated metadata.
+
+    Returns:
+        The replicated file to be created in the destination.
+    """
+    logging.debug(f"Creating a new file based on source file id {src_file.id}")
+    mime_type = mimetypes.types_map.get(f".{src_file.mime_type}", src_file.mime_type)
+
+    return FileMetadata(
+        external_id=src_file.external_id,
+        name=src_file.name,
+        source=src_file.source,
+        mime_type=mime_type,
+        metadata=replication.new_metadata(src_file, project_src, runtime),
+        asset_ids=replication.get_asset_ids(src_file.asset_ids, src_dst_ids_assets),
+    )
+
+
+def update_file(
+    src_file: FileMetadata, dst_file: FileMetadata, src_dst_ids_assets: Dict[int, int], project_src: str, runtime: int
+) -> FileMetadata:
+    """
+    Makes an updated version of the destination file based on the corresponding source file.
+
+    Args:
+        src_file: The file from the source to be replicated.
+        dst_file: The file from the destination that needs to be updated to reflect changes made to its source file.
+        src_dst_ids_assets: A dictionary of all the mappings of source asset id to destination asset id.
+        project_src: The name of the project the object is being replicated from.
+        runtime: The timestamp to be used in the new replicated metadata.
+
+    Returns:
+        The updated file object for the replication destination.
+    """
+    logging.debug(f"Updating existing file {dst_file.id} based on source file id {src_file.id}")
+
```

Comment:
Is mime_type immutable?

Reply:
Per https://docs.cognite.com/api/v1/#operation/updateFiles it looks like that is the case

Reply:
Interesting :)